File: Safari-611.1.10/LayoutTests/http/tests/security/contentSecurityPolicy/resources/worker-xhr-redirect-cross-origin-blocked.php
<?php
    require "determine-content-security-policy-header.php";
?>
var expectedExceptionCode = 19; // DOMException.NETWORK_ERR
var isAsynchronous = false;
var xhr = new XMLHttpRequest;
try {
    xhr.open("GET", "http://127.0.0.1:8000/security/contentSecurityPolicy/resources/redir.php?url=http://localhost:8000/xmlhttprequest/resources/access-control-basic-allow.cgi", isAsynchronous);
    xhr.send();
    self.postMessage("FAIL should throw " + expectedExceptionCode + ". But did not throw an exception.");
} catch (exception) {
    if (exception.code === expectedExceptionCode)
        self.postMessage("PASS threw exception " + exception + ".");
    else
        self.postMessage("FAIL should throw " + expectedExceptionCode + ". Threw exception " + exception + ".");
}

File: LayoutTests/http/tests/security/contentSecurityPolicy/resources/worker-xhr-redirect-cross-origin-blocked.py
#!/usr/bin/env python3

import sys
from utils import determine_content_security_policy_header

determine_content_security_policy_header()
sys.stdout.write(
    'Content-Type: text/javascript\r\n\r\n'
    'var expectedExceptionCode = 19; // DOMException.NETWORK_ERR\n'
    'var isAsynchronous = false;\n'
    'var xhr = new XMLHttpRequest;\n'
    'try {\n'
    '    xhr.open("GET", "http://127.0.0.1:8000/security/contentSecurityPolicy/resources/redir.py?url=http://localhost:8000/xmlhttprequest/resources/access-control-basic-allow.cgi", isAsynchronous);\n'
    '    xhr.send();\n'
    '    self.postMessage("FAIL should throw " + expectedExceptionCode + ". But did not throw an exception.");\n'
    '} catch (exception) {\n'
    '    if (exception.code === expectedExceptionCode)\n'
    '        self.postMessage("PASS threw exception " + exception + ".");\n'
    '    else\n'
    '        self.postMessage("FAIL should throw " + expectedExceptionCode + ". Threw exception " + exception + ".");\n'
    '}\n'
)
